File: mobile/lib/services/auth_service.dart
import 'dart:convert';
import 'package:shared_preferences/shared_preferences.dart';
import 'api_service.dart';
import '../models/user.dart';
import '../utils/constants.dart';

class AuthService {
  final ApiService _api = ApiService();

  Future<void> sendOTP(String email) async {
    await _api.post('/auth/send-otp', body: {'email': email});
  }

  Future<User> verifyOTP(String email, String otp) async {
    final response = await _api.post('/auth/verify-otp', body: {
      'email': email,
      'otp': otp,
    });

    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(Constants.tokenKey, response['token']);
    await prefs.setString(Constants.userKey, jsonEncode(response['user']));

    return User.fromJson(response['user']);
  }

  static Future<User?> getCurrentUser() async {
    final prefs = await SharedPreferences.getInstance();
    final userJson = prefs.getString(Constants.userKey);
    if (userJson != null) {
      return User.fromJson(jsonDecode(userJson));
    }
    return null;
  }

  static Future<String?> getToken() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString(Constants.tokenKey);
  }

  static Future<void> logout() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove(Constants.tokenKey);
    await prefs.remove(Constants.userKey);
  }

  static Future<bool> isLoggedIn() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.containsKey(Constants.tokenKey);
  }
}

File: frontend/src/services/authService.js
import api from './api';

export const authService = {
  sendOTP: async (email) => {
    return api('/auth/send-otp', {
      method: 'POST',
      body: JSON.stringify({ email }),
    });
  },

  verifyOTP: async (email, otp) => {
    const response = await api('/auth/verify-otp', {
      method: 'POST',
      body: JSON.stringify({ email, otp }),
    });

    console.log('📨 Verify OTP response:', response);

    if (response.token && response.user) {
      // ✅ Ensure the user object is stored correctly
      localStorage.setItem('token', response.token);
      localStorage.setItem('user', JSON.stringify(response.user));
      console.log('💾 Stored user with role:', response.user.role);
    } else {
      console.warn('⚠️ Missing token or user in response');
    }

    return response;
  },

  getCurrentUser: async () => {
    return api('/auth/me', { method: 'GET' });
  },

  logout: () => {
    localStorage.removeItem('token');
    localStorage.removeItem('user');
    console.log('🗑️ Local storage cleared');
  },

  isAuthenticated: () => {
    const token = localStorage.getItem('token');
    return !!token;
  },

  getUser: () => {
    const user = localStorage.getItem('user');
    if (user) {
      try {
        const parsed = JSON.parse(user);
        console.log('📖 Retrieved user from storage:', parsed);
        return parsed;
      } catch (e) {
        console.error('❌ Error parsing user:', e);
        return null;
      }
    }
    return null;
  },

  getToken: () => {
    const token = localStorage.getItem('token');
    console.log('🔑 Retrieved token:', token ? 'exists' : 'none');
    return token;
  },
};